main.go

```diff
@@ -117,8 +118,11 @@ func loadConfig(p string) error {
 	if config.Server == "" {
 		return fmt.Errorf("no server specified")
 	}
-	if len(config.Chans) < 1 {
-		return fmt.Errorf("no channels specified")
+	if len(config.Chans) < 1 && len(config.Feeds) < 1 {
+		return fmt.Errorf("no channels specified (need either chans or feeds)")
+	}
+	for _, c := range config.Chans {
+		joinedChans[c] = true
```

Comment:
I think a name like `prejoinedChans` would be clearer. I initially read this code as "channels we've temporarily joined", not as "channels we join at startup and stay until exit".